modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4575,8 +4575,14 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
         }
 
         Schema parameterSchema;
+
+        // the parameter model name is obtained from the schema $ref
+        // e.g. #/components/schemas/list_pageQuery_parameter => toModelName(list_pageQuery_parameter)
+        String parameterModelName = null;
+
         if (parameter.getSchema() != null) {
-            parameterSchema = parameter.getSchema();
+            parameterModelName = getParameterDataType(parameter ,parameter.getSchema());
+            parameterSchema = ModelUtils.getReferencedSchema(openAPI, parameter.getSchema());
```

Comment:
It looks like the generators are fighting eachother on the assignment of:
- parameter.dataType
- parameter.complexType
- parameter.schema
How about setting custom methods to set those for each generator so each can do what it needs?

Reply:
Something odd is also happening with unaliasSchema
The original code extracts the schema, unaliases it, then uses that unaliased schemas to set properties.

This code instead gets the schema from the ref automatically. 
So why is unaliasSchema not handling this case automatically originally?

Reply:
Please see my branch to fix the python-experimental issue:
https://github.com/OpenAPITools/openapi-generator/pull/12397/files

unaliasSchema is not retuning the actual enum schema because if an schema contains a ref to an enum schema, unaliasSchmema returns that schema with the $ref in it. That assumes that the generator will generate a model reference to the enum schema.
- One could limit `ModelUtils.getReferencedSchema` to only run when the ref is to an enum schema
- OR one could make sure that the import and type reference is generator for this inline schema rather than inlining the enum definition which typescript is doing